src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -1034,6 +1034,8 @@ impl<'tcx> LocationTree {
             wildcard_state.access_relatedness(access_kind, only_foreign)
         };
 
+        let mut has_exposed = false;
```

Comment:
This isn't just about whether there is an exposed node, it's about whether there's an exposed node *that allows this access*. The name should at least indicate that, and there should be a comment here explaining that.